File: BackEnd/src/domain/interfaces/usecases/chat/IMarkMessagesAsReadUseCase.ts
/**
 * @file IMarkMessagesAsReadUseCase.ts
 * @description Interfaz del caso de uso para marcar mensajes como leídos.
 * @module Chat
 * @layer Domain
 */
export interface IMarkMessagesAsReadUseCase {
  execute(conversationId: string, userId: string): Promise<void>;
}

File: FrontEnd/lib/domain/interfaces/usecases/chat/i_mark_messages_as_read_use_case.dart
/// @file i_mark_messages_as_read_use_case.dart
/// @description Interfaz: Marca como leídos todos los mensajes de una conversación.
/// @module Chat
/// @layer Domain
library;
abstract interface class IMarkMessagesAsReadUseCase {
  /// Marca como leídos todos los mensajes de una conversación.
  Future<void> execute(String conversationId);
}
